packages/bson-transpilers/codegeneration/javascript/Visitor.js

```diff
@@ -604,9 +604,7 @@ module.exports = (CodeGenerationVisitor) => class Visitor extends CodeGeneration
       Buffer: Buffer,
       __result: {}
     };
-    const ctx = new Context(sandbox);
-    const res = ctx.evaluate('__result = ' + input);
-    ctx.destroy();
+    const res = vm.runInContext('__result = ' + input, vm.createContext(sandbox));
```

Comment:
On node this is all that [context-eval](https://github.com/amasad/context-eval) does. For the browser (ie. data explorer) the plan is to use a [polyfill](https://github.com/browserify/vm-browserify).

context-eval broke for us in this upgrade for kinda tricky reasons:

Before this package was using jsdom via [jsdom-global](https://github.com/rstacruz/jsdom-global) which has a [hardcoded list of properties](https://github.com/rstacruz/jsdom-global/blob/master/keys.js) to copy over to the global (window) object. The shared config (ie. the new way) uses [global-jsdom](https://github.com/modosc/global-jsdom) which uses [Object.getOwnPropertyNames(window)](https://github.com/modosc/global-jsdom/blob/5b90b116426869054cfd551daa7d83b733a35395/commonjs/index.js#L38).

For whatever reason bson-transpiler executes some of the javascript it transpiles and it does this in a context. That requires the jsdom `runScripts: "dangerously"` option to be set. Unfortunately this causes (for reasons we're not 100% clear on, I think) `Object.getOwnPropertyNames(window)` to not return things like `Element` which breaks leafygreen.

So to recap: Without runScripts: "dangeriously" tests would fail with ObjectId not defined. With it leafygreen would fail.

Looking at context-eval, it looks for the existance of the `window` object to know if it is running in node (where it can use the vm module) or in the browser (where it falls back to using an iframe). Since the tests are running in node could just be using `vm`. In electron we should have access to it as well. Elsewhere we shouldn't probably be polyfilling it.

So the cleanest solution seems to be to just not use context-eval and polyfill node's vm module if necessary.

Reply:
I may be missing something do we already have the polyfill in this PR? Does something change for Data Explorer?

Reply:
We don't seem to need the polyfill at this point because electron seems to make it work for us. Data Explorer will have to add it. We're testing to see if it will work.

Btw we might eventually need the polyfill as well if we make the changes to not rely on electron and use the same stuff they do.